### PR TITLE
Added Filters for Snapshot Datasource

### PR DESCRIPTION
**Pull request by akshay-sripriya**
# Description
Added support for all query parameters for Snapshot Datasource.
Added Name filter for Snapshot Datasource

##### RESOURCE OR DATASOURCE NAME

Snapshot Datasource

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by akshay-sripriya** [comment]
Screenshot 2024-10-28 112420

**Comment by akshay-sripriya** [comment]
run e2e test SnapshotDataSource

**Comment by akshay-sripriya** [comment]
run e2e test SnapshotDataSource

**Comment by Krunal-Thakkar** [comment]
Add code coverage report

**Review by Krunal-Thakkar — APPROVED** [review]

**Comment by Krunal-Thakkar** [comment]
Does we require any change in Unit Test side?

**Comment by akshay-sripriya** [comment]
run e2e test SnapshotDataSource

File: powerscale/provider/snapshot_datasource.go [hunk 1]
```diff
@@ -158,6 +158,46 @@ func (d *SnapshotDataSource) Schema(ctx context.Context, req datasource.SchemaRe
 					"path": schema.StringAttribute{
 						Optional: true,
 					},
+
+					"name": schema.StringAttribute{
```
Comment (Sakshi-dell) [inline powerscale/provider/snapshot_datasource.go:162]: path and name should be mutually exclusive right? what if user tries to give both?
Also, these should be mutually exclusive with all other filters like sort, limit and so on

File: powerscale/provider/snapshot_datasource.go [hunk 2]
```diff
@@ -238,12 +278,31 @@ func (d *SnapshotDataSource) Read(ctx context.Context, req datasource.ReadReques
...
+		for _, sdm := range fulldetail {
+			if plan.SnapshotFilter.Name.ValueString() == sdm.Name.ValueString() {
+				state.Snapshots = append(state.Snapshots, sdm)
+			}
+		}
+		// If after the filter the length is still zero then that filter is invalid
+		if len(state.Snapshots) == 0 {
+			resp.Diagnostics.AddError(
+				"Error getting snapshots",
+				fmt.Sprintf("Name `%s` is invalid, it has no snapshots ", plan.SnapshotFilter.Path.ValueString()),
```
Comment (shenda1) [inline powerscale/provider/snapshot_datasource.go:293]: I think the better message would be "Unable to find snapshot with name %s", plan.SnapshotFilter.Name.ValueString"

Comment (akshay-sripriya) [reply powerscale/provider/snapshot_datasource.go]: Changed this in the recent commit

**Comment by akshay-sripriya** [comment]
run e2e test SnapshotDataSource

**Comment by Sakshi-dell** [comment]
run e2e test SnapshotDataSource

File: powerscale/provider/snapshot_datasource_test.go [hunk 3]
```diff
@@ -102,6 +119,18 @@ var SnapshotDataSourceConfig = `
 data "powerscale_snapshot" "test" {
   filter {
     path = "/ifs/tfacc_file_system_test"
+	state = "active"
```
Comment (Sakshi-dell) [inline powerscale/provider/snapshot_datasource_test.go:122]: same for this as well, like creating a snapshot that exists on this path and then fetch the same using datasource

**Comment by akshay-sripriya** [comment]
run e2e test SnapshotDataSource

**Comment by akshay-sripriya** [comment]
run e2e test SnapshotDataSource

**Comment by akshay-sripriya** [comment]
run e2e test SnapshotDataSource

**Comment by akshay-sripriya** [comment]
Screenshot 2024-10-30 154243

**Comment by akshay-sripriya** [comment]
run e2e test SnapshotDataSource

**Comment by Sakshi-dell** [comment]
run e2e test SnapshotDataSource

**Comment by Krunal-Thakkar** [comment]
run e2e test SnapshotDataSource

**Comment by akshay-sripriya** [comment]
run e2e test SnapshotDataSource

**Review by Sakshi-dell — APPROVED** [review]

**Pull request merged**